examples/docker/kafka-oauth-strimzi/kafka/oauth.sh

```diff
@@ -0,0 +1,119 @@
+#!/bin/bash
+
+usage() {
+  echo "Usage: $0 [USERNAME] [PASSWORD] [ARGUMENTS] ..."
+  echo
+  echo " USERNAME    The username for user authentication"
+  echo " PASSWORD    The password for user authentication (prompted for if not specified)"
+  echo
+  echo " If USERNAME and PASSWORD are not specified, client credentials will be used for authentication."
+  echo
+  echo " ARGUMENTS:"
+  echo "   --quiet, -q                      No informational outputs"
+  echo "   --insecure                       Allow http:// in token endpoint url"
+  echo "   --access                         Return access_token rather than refresh_token"
+  echo "   --endpoint  TOKEN_ENDPOINT_URL   Authorization server token endpoint"
+  echo "   --client-id CLIENT_ID            Client id for client authentication - must be configured on authorization server"
+  echo "   --secret    CLIENT_SECRET        Secret to authenticate the client"
+  echo "   --scopes    SCOPES               Space separated list of scopes to request - default value: offline_access"
+}
+
+
+CLAIM=refresh_token
+GRANT_TYPE=password
+DEFAULT_SCOPES=offline_access
+
+while [ $# -gt 0 ]
+do
+    case "$1" in
+      "-q" | "--quiet")
+          QUIET=1
+          ;;
+      --endpoint)
+          shift
+          TOKEN_ENDPOINT="$1"
+          ;;
+      --insecure)
+          INSECURE=1
+          ;;
+      --access)
+          CLAIM=access_token
+          DEFAULT_SCOPES=""
+          ;;
+      --client-id)
+          shift
+          CLIENT_ID="$1"
+          ;;
+      --secret)
+          shift
+          CLIENT_SECRET="$1"
+          ;;
+      --scopes)
+          shift
+          SCOPES="$1"
+          ;;
+      --help)
+          usage
+          exit 1
+          ;;
+      *)
+          if [ "$UNAME" == "" ]; then
+            UNAME="$1"
+          elif [ "$PASS" == "" ]; then
+            PASS="$1"
+          else
+            >&2 echo "Unexpected argument!"
+            exit 1
+          fi
+          ;;
+    esac
+    shift
+done
+
+if [ "$TOKEN_ENDPOINT" == "" ]; then
+    >&2 echo "ENV variable TOKEN_ENDPOINT not set."
+    exit 1
+fi
+
+if [ "$UNAME" != "" ] && [ "$PASS" == "" ]; then
+    >&2 read -s -p "Password: " PASS
+    >&2 echo
+fi
+
+if [ "$UNAME" == "" ] && [ "CLIENT_ID" == "" ]; then
+    echo "Usage: $0 [USERNAME] [PASSWORD] [--insecure] [--access] [--endpoint TOKEN_ENDPOINT_URL] [--client-id CLIENT_ID] [--secret CLIENT_SECRET] [--scopes SCOPES]"
```

Comment:
Say that both USERNAME and --client-id must be given

Reply:
That's not actually the case - the --client-id only has to be specified if USERNAME has not been, otherwise it default to 'kafka-cli' value. The error here is actually the missing '$' in "CLIENT_ID"